### PR TITLE
💄 #512: consent step 1 API hookup

### PR DESCRIPTION
**Pull request by samrichca**
### Consent UI
- add a LoadingOverlay component
  - temporary measure. adding other loading animations to the consent wizard would require refactoring.
- fetch invite data on load
- submit invite data on form submit
- add suspense boundary to consent wizard

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/consent-ui/src/components/views/Register/handleGetInvite.tsx [hunk 1]
```diff
@@ -17,8 +17,6 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-'use server';
```
Comment (samrichca) [inline apps/consent-ui/src/components/views/Register/handleGetInvite.tsx:20]: removing 'use server' for things that don't need to be server actions

**Comment by samrichca** [comment]
check if the email client works when you click on the mailto links at the top and the bottom @anncatton

**Review by anncatton — COMMENTED** [review]
Looks good! Just some minor comments on the `server-only` stuff and the early return on no session

**Review by anncatton — APPROVED** [review]
Thanks for starting a clear naming convention for these requests. Looks good to go 🚀

**Review by anncatton — APPROVED** [review]

**Pull request merged**